src/pdf_parser/pdf_utils/disambiguator/overlapping.py

```diff
@@ -0,0 +1,235 @@
+import itertools
+from typing import Optional, Tuple
+
+from layoutparser import Layout, TextBlock, Rectangle
+
+from src.pdf_parser.pdf_utils.disambiguator.utils import is_in
+
+
+def horizontal_overlap(box_1: TextBlock, box_2: TextBlock) -> bool:
+    """Check if the boxes overlap horizontally."""
+    if (
+        box_1.coordinates[0] < box_2.coordinates[2]
+        and box_1.coordinates[2] > box_2.coordinates[0]
+    ):
+        return True
+    return False
+
+
+def vertical_overlap(box_1: TextBlock, box_2: TextBlock) -> bool:
+    """Check if the boxes overlap vertically."""
+    if (
+        box_1.coordinates[1] < box_2.coordinates[3]
+        and box_1.coordinates[3] > box_2.coordinates[1]
+    ):
+        return True
+    return False
```

Comment:
Can we add some comments or docstring details to explain the numbers in these functions? I know it's documented elsewhere but the coordinates are not intutitive to someone just looking at it.